packages/starknet-snap/src/index.ts

```diff
@@ -124,6 +125,10 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ origin, request }) =>
         apiParams.keyDeriver = await getAddressKeyDeriver(snap);
         return createAccount(apiParams);
 
+      case 'starkNet_createAccountLegacy':
+        apiParams.keyDeriver = await getAddressKeyDeriver(snap);
+        return createAccount(apiParams, false, true, CAIRO_VERSION_LEGACY);
```

Comment:
i think we dont need this api, can we put the cairo version into the apiParams
so we can re use starkNet_createAccount?
and we can also put 
silentMode = false,
  waitMode = true in starkNet_createAccount